profiles/models.py

```diff
@@ -1,3 +1,447 @@
+            MinValueValidator(1900),
+            MaxValueValidator(datetime.datetime.now().year)
+        ]
+    )
+    team_size = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        validators=[MinValueValidator(1)]
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class Startup(Company):
+    user = models.OneToOneField(
+        'users.User',
+        on_delete=models.CASCADE,
+        related_name='startup'
+    )
+    industry = models.ForeignKey(
+        'Industry',
+        on_delete=models.PROTECT,
+        related_name='startups'
+    )
+    location = models.ForeignKey(
+        'Location',
+        on_delete=models.PROTECT,
+        related_name='startup_locations'
+    )
+    startup_logo = models.ImageField(upload_to='startups/logos/')
+    social_media = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        validators=[MaxLengthValidator(255)]
+    )
+    stage = models.CharField(
+        max_length=20,
+        choices=Stage.choices,
+        blank=True,
+        null=True
+    )
+
+    def __str__(self):
+        return f"{self.company_name} (Startup, User ID: {self.user_id})"
+
+    class Meta:
+        db_table = "startups"
+        ordering = ["company_name"]
+        verbose_name = "Startup"
+        verbose_name_plural = "Startups"
+        constraints = [
+            models.UniqueConstraint(fields=["user"], name="unique_startup_user")
+        ]
+
+
+class Investor(Company):
+    user = models.OneToOneField(
+        'users.User',
+        on_delete=models.CASCADE,
+        related_name='investor'
+    )
+    industry = models.ForeignKey(
+        'Industry',
+        on_delete=models.PROTECT,
+        related_name='investors'
+    )
+    location = models.ForeignKey(
+        'Location',
+        on_delete=models.PROTECT,
+        related_name='investor_locations'
+    )
+    investor_logo = models.ImageField(upload_to='investors/logos/')
+    interests = models.TextField(blank=True, null=True)
+    fund_size = models.DecimalField(
+        max_digits=20,
+        decimal_places=2,
+        blank=True,
+        null=True,
+        validators=[MinValueValidator(0)]
+    )
+    country = models.ForeignKey(
+        'Country',
+        on_delete=models.PROTECT,
+        related_name='investors'
+    )
+    preferred_stage = models.CharField(
+        max_length=20,
+        choices=Stage.choices,
+        blank=True,
+        null=True
+    )
+
+    def __str__(self):
+        return f"{self.company_name} (Investor, User ID: {self.user_id})"
+
+    class Meta:
+        db_table = "investors"
+        ordering = ["company_name"]
+        verbose_name = "Investor"
+        verbose_name_plural = "Investors"
+        constraints = [
+            models.UniqueConstraint(fields=["user"], name="unique_investor_user")
+        ]
+
```

Comment:
These two classes share a similar structure (logo, industry, location, user link, etc.).
Consider an additional intermediate abstract class, e.g., class ProfiledCompany(Company) for shared fields like industry, location, logo.